test/e2e/network_test.go

```diff
@@ -567,13 +595,28 @@ func (tc *testContext) waitUntilJobSucceeds(name string) error {
 			return nil
 		}
 		if job.Status.Failed > 0 {
-			events, _ := tc.getPodEvents(name)
-			return errors.Errorf("job %v failed: %v", job, events)
+			tc.printJobLogsAndEvents(job)
```

Comment:
Shouldn't we be printing this for all failures where `Jobs` / `Deployments` are involved? Like in the east-west networking case also? This should be easy to do given we can make `getLogs()` take a selector as input. 

@sebsoto the other question I have is for you: What is the main use case for this? Don't these logs anyhow show up in the artifacts folder in the case of a test failure as `wmco-test` namespace is not deleted? Or am I missing something?

Reply:
Yeah I could add that if we think it would be useful to print pod logs from a deployment failure too. I could change getPodLogs() to also take a selector as input and then add a printJobLogsAndEvents() call in waitUntilDeploymentScaled(). Should I do that in this PR or in a followup PR?

Reply:
>I could change getPodLogs() to also take a selector as input

To be clear, it would need **just** the selector

> and then add a printJobLogsAndEvents() call in waitUntilDeploymentScaled().

Not sure if this is correct spot given we want to see the logs on all test failures. The deployment not scaling is just one of the scenarios.

>Should I do that in this PR or in a followup PR?

It can be done in this PR but I would hold off on making the change until we hear back from @sebsoto.

Reply:
If its desirable to get the logs from deployment related pods as well I'm ok with that

Reply:
>At the time that I wrote this ticket up that was not the case, and I still think that it is inconsistent.

Do you have an example of this inconsistency? I have always found the logs in the artifacts folder when a test failed. It is only on success that I have not seen the logs.

>This is actually being used to debug the east-west networking case. It is gathering logs for the pod that is attempting to GET from the webserver. The response that the getter receives is what provides us the most value in the networking case because we get to see at what point the GET fails, and if it is getting a response code at all.
If its desirable to get the logs from deployment related pods as well I'm ok with that

If we are doing this we should do this generically for all cases and not just a specific test. Moreover all these logs should be posted to a file in the artifacts folder similar to what we are doing with the WMCO logs. We should also consider if we want to get the logs on success also.

Reply:
> > and then add a printJobLogsAndEvents() call in waitUntilDeploymentScaled().
> 
> Not sure if this is correct spot given we want to see the logs on all test failures. The deployment not scaling is just one of the scenarios.
> 
I mean it could be called there in addition to being called in waitUntilJobSucceeds(). Having it called in both with a selector as input would then cover all failures where Jobs / Deployments are involved.